sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -676,7 +676,9 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     object ExtractAttribute {
       def unapply(expr: Expression): Option[Attribute] = {
         expr match {
-          case attr: Attribute => Some(attr)
+          case attr: Attribute
+              if attr.dataType == StringType || IntegralType.acceptsType(attr.dataType) =>
```

Comment:
you may define a private function for this.

Reply:
Sounds good, I'll update for that.